### ISSUE TITLE
Injection plugin not working

### ISSUE THREAD

**Issue by DX37** [ventoy/Ventoy#343]
Used archives _zip_ and _7z_ with _Normal_ compress mode. Applied these archives to _Windows PE by Sergei Strelec_ and to _Windows 10 2004 iso_. Used this configuration:
```
{
    "control": [
        { "VTOY_DEFAULT_MENU_MODE": "1" }
    ],

    "auto_install" : [
        {
            "image": "/Windows/Win10_2004_Russian_x64.iso",
            "template": [
                "/ventoy/win10/uefi/home.xml",
                "/ventoy/win10/uefi/home_single.xml",
                "/ventoy/win10/uefi/pro.xml",
                "/ventoy/win10/bios/home.xml",
                "/ventoy/win10/bios/home_single.xml",
                "/ventoy/win10/bios/pro.xml"
            ]
        },
        {
            "image": "/Windows/Win10_1909_Russian_x64.iso",
            "template": [
                "/ventoy/win10/uefi/home.xml",
                "/ventoy/win10/uefi/home_single.xml",
                "/ventoy/win10/uefi/pro.xml",
                "/ventoy/win10/bios/home.xml",
                "/ventoy/win10/bios/home_single.xml",
                "/ventoy/win10/bios/pro.xml"
            ]
        },
        {
            "image": "/Windows/Win7_SP1_Russian_x64.iso",
            "template": [
                "/ventoy/win7/uefi/home_basic.xml",
                "/ventoy/win7/uefi/home_premium.xml",
                "/ventoy/win7/uefi/pro.xml",
                "/ventoy/win7/bios/home_basic.xml",
                "/ventoy/win7/bios/home_premium.xml",
                "/ventoy/win7/bios/pro.xml"
            ]
        }
    ],
    "injection": [
        {
            "image": "/Windows/Win10_2004_Russian_x64.iso",
            "archive": "/ventoy/Desktop.7z"
        },
        {
            "image": "/WinPE10_8_Sergei_Strelec_x86_x64_2020.05.31_Russian.iso",
            "archive": "/ventoy/Desktop.7z"
        }
    ]
}
```
None of the isos didn't show any new files under **X:** directory.

Is this thing working? Or I am doing it wrong?

**Comment by ventoy** [comment]
1. Use F5 to check the injection configuration
2. Check  if there is  a **X:\windows\System32\ventoy.log**  and dump it.

**Comment by DX37** [comment]
> 
> 
>     1. Use F5 to check the injection configuration
> 
>     2. Check  if there is  a **X:\windows\System32\ventoy.log**  and dump it.

Injection config is correct, even in VM.
![изображение](https://user-images.githubusercontent.com/12829428/88381719-1f95ee80-cdd1-11ea-9dca-206893bd60be.png)

Got it. Something with filesystem. I have exFAT on my main partition. Right now I can't test in another filesystem, since I got that partition almost full with necessary files.
```
[2020/07/24 17:14:46.809] ######## VentoyJump ##########
[2020/07/24 17:14:46.809] argc = 1 argv[0] = <X:\windows\system32\winpeshl.exe>
[2020/07/24 17:14:46.825] Current directory = <X:\windows\system32>
[2020/07/24 17:14:46.825] ReadWholeFile2Buf <X:\windows\system32\winpeshl.exe>
[2020/07/24 17:14:46.825] Success read file size:161280
[2020/07/24 17:14:46.825] Find os pararm at 127488
[2020/07/24 17:14:46.825] SaveBuffer2File <ventoy\winpeshl.exe> len:32256
[2020/07/24 17:14:46.825] Logical Drives=0x80000c Path:<\WinPE10_8_Sergei_Strelec_x86_x64_2020.05.31_Russian.iso>
[2020/07/24 17:14:46.825] File exist under C:
[2020/07/24 17:14:46.825] GetPhyDiskUUID C
[2020/07/24 17:14:46.825] C: is in PhysicalDrive0 
[2020/07/24 17:14:46.825] Disk UUID match
[2020/07/24 17:14:46.825] Find ISO file <C:\\WinPE10_8_Sergei_Strelec_x86_x64_2020.05.31_Russian.iso>
[2020/07/24 17:14:46.825] This is Windows 8 or latter...
[2020/07/24 17:14:46.825] VentoyMountISOByAPI <C:\\WinPE10_8_Sergei_Strelec_x86_x64_2020.05.31_Russian.iso>
[2020/07/24 17:14:46.887] OpenVirtualDisk success
[2020/07/24 17:14:46.903] Mount iso by API success
[2020/07/24 17:14:46.903] Mount ISO FILE: SUCCESS
[2020/07/24 17:14:46.903] DeleteVentoyPart2MountPoint Phy0 ...
[2020/07/24 17:14:46.935] Delete ventoy mountpoint: NO NEED
[2020/07/24 17:14:46.935] auto install no need
[2020/07/24 17:14:46.935] decompress injection archive C:/ventoy/Desktop.7z...
[2020/07/24 17:14:46.935] DecompressInjectionArchive C:/ventoy/Desktop.7z
[2020/07/24 17:14:46.935] Parse FAT fs...
[2020/07/24 17:14:46.950] FAT_FS: Error could not load FAT details (-3)!


[2020/07/24 17:14:46.950] Ventoy jump success ...

```

**Comment by ventoy** [comment]
I just find a bug in the code,  the injection will fail if you have reserved some space at the bottom of the disk.
Is this the case for you?

**Comment by DX37** [comment]
> I just find a bug in the code,  the injection will fail if you have reserved some space at the bottom of the disk.
> Is this the case for you?
> 

No, not at all. I used GPT install of Ventoy with Secure Boot and that's all.
Is it normal if I have almost 59 GB partition in exFAT filesystem?

**Comment by ventoy** [comment]
Yes, GPT will have this bug too.  I will fix it soon.



    


            On 07/24/2020 19:58, Maxim Kurnosenko wrote: 

I just find a bug in the code,  the injection will fail if you have reserved some space at the bottom of the disk.
Is this the case for you?

No, not at all. I used GPT install of Ventoy with Secure Boot and that's all.
Is it normal if I have almost 59 GB partition in exFAT filesystem?

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/ventoy/Ventoy/issues/343#issuecomment-663503775",
"url": "https://github.com/ventoy/Ventoy/issues/343#issuecomment-663503775",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by ventoy** [comment]
@DX37 
I fixed the bug, you can update and retry.

You need to update the two files in the 2nd partition (VTOYEFI partition) .

`ventoy\vtoyjump32.exe`
`ventoy\vtoyjump64.exe`

from  [https://github.com/ventoy/Ventoy/tree/master/INSTALL/ventoy](https://github.com/ventoy/Ventoy/tree/master/INSTALL/ventoy)

**Comment by DX37** [comment]
@ventoy Got it, I'll test this now.

**Comment by DX37** [comment]
@ventoy Yep, now it fully works, I guess. At least with archives I told you about in the first post.
Thank you!

**Comment by DX37** [comment]
fixed by v1.0.17
